src/policy/markcompactspace.rs

```diff
@@ -128,6 +128,10 @@ impl<VM: VMBinding> crate::policy::gc_work::PolicyTraceObject<VM> for MarkCompac
         _copy: Option<CopySemantics>,
         _worker: &mut GCWorker<VM>,
     ) -> ObjectReference {
+        assert!(
```

Comment:
This is performance critical code. So use `debug_assert` or `static_assertions` instead.